orchagent/fdborch.cpp

```diff
@@ -118,85 +128,233 @@ void FdbOrch::update(sai_fdb_event_t type, const sai_fdb_entry_t* entry, sai_obj
+        }
+
+        // We should already have such entry
+        if (existing_entry == m_entries.end())
+        {
+             SWSS_LOG_INFO("FdbOrch MOVE notification: mac %s is not found in bv_id 0x%lx",
+                    update.entry.mac.to_string().c_str(), entry->bv_id);
+             break;
+        }
+        if (!m_portsOrch->getPortByBridgePortId(existing_entry->second.bridge_port_id, port_old))
+        {
+            SWSS_LOG_ERROR("FdbOrch MOVE notification: Failed to get port by bridge port ID 0x%lx", existing_entry->second.bridge_port_id);
+            return;
+        }
+        port_old.m_fdb_count--;
+        m_portsOrch->setPort(port_old.m_alias, port_old);
+        update.port.m_fdb_count++;
+        m_portsOrch->setPort(update.port.m_alias, update.port);
+
+        storeFdbEntryState(update);
+
+        for (auto observer: m_observers)
+        {
+            observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
+        }
+        break;
+    }
     case SAI_FDB_EVENT_FLUSHED:
-        if (bridge_port_id == SAI_NULL_OBJECT_ID && entry->bv_id == SAI_NULL_OBJECT_ID)
+        SWSS_LOG_INFO("Received FLUSH event for bvid=%lx mac=%s port=%lx", entry->bv_id, update.entry.mac.to_string().c_str(), bridge_port_id);
+        for (auto itr = m_entries.begin(); itr != m_entries.end();)
         {
-            for (auto itr = m_entries.begin(); itr != m_entries.end();)
+            if (itr->second.type == "static")
             {
-                /*
-                   TODO: here should only delete the dynamic fdb entries,
-                   but unfortunately in structure FdbEntry currently have
-                   no member to indicate the fdb entry type,
-                   if there is static mac added, here will have issue.
-                */
-                update.entry.mac = itr->mac;
-                update.entry.bv_id = itr->bv_id;
+                itr++;
+                continue;
+            }
+
+            if (((bridge_port_id == SAI_NULL_OBJECT_ID) && (entry->bv_id == SAI_NULL_OBJECT_ID)) // Flush all DYNAMIC
+                || ((bridge_port_id == itr->second.bridge_port_id) && (entry->bv_id == SAI_NULL_OBJECT_ID)) // flush all DYN on a port
+                || ((bridge_port_id == SAI_NULL_OBJECT_ID) && (entry->bv_id == itr->first.bv_id))) // flush all DYN on a vlan
+            {
+
+                if (!m_portsOrch->getPortByBridgePortId(itr->second.bridge_port_id, update.port))
+                {
+                    SWSS_LOG_ERROR("FdbOrch FLUSH notification: Failed to get port by bridge port ID 0x%lx", itr->second.bridge_port_id);
+                    itr++;
+                    continue;
+                }
+
+                if (!m_portsOrch->getPort(itr->first.bv_id, vlan))
+                {
+                    SWSS_LOG_NOTICE("FdbOrch FLUSH notification: Failed to locate vlan port from bv_id 0x%lx", itr->first.bv_id);
+                    itr++;
+                    continue;
+                }
```

Comment:
Even though the vlan or port is not exist, this MAC still need to be deleted.

Reply:
We dont delete the bridge port or the vlan until all mac are deleted and port or vlan fdb count is zero. so, it is not expected for the bridge port or vlan to be not there.